test/gtest/uct/test_atomic_key_reg_rdma_mem_type.cc

```diff
@@ -30,14 +30,41 @@ UCS_TEST_SKIP_COND_P(uct_atomic_key_reg_rdma_mem_type, fadd64,
                      !check_atomics(UCS_BIT(UCT_ATOMIC_OP_ADD), FOP64) ||
                      !check_rdma_memory())
 {
-    mapped_buffer recvbuf(sizeof(uint64_t), receiver(), 0UL,
-                          UCS_MEMORY_TYPE_RDMA);
+    mapped_buffer *recvbuf     = nullptr;
+    const size_t buffer_size   = sizeof(uint64_t);
+    const size_t buffer_offset = 0;
+    const bool may_fail        = true;
+    const int num_retries      = 5;
+    const int sleep_usec       = 10000;
+
+    for (int i = 0; i < num_retries; ++i) {
+        try {
+            {
+                scoped_log_handler slh(hide_errors_logger);
+                recvbuf = new mapped_buffer(buffer_size, receiver(),
+                                            buffer_offset, UCS_MEMORY_TYPE_RDMA,
+                                            UCT_MD_MEM_ACCESS_ALL, may_fail);
+            }
+            break;
+        } catch (ucs::test_abort_exception& e) {
+            UCS_TEST_MESSAGE << "Retry " << i + 1 << "/" << num_retries
+                             << ": Buffer allocation failed - " << e.what();
+            usleep(sleep_usec);
+        }
+
+        if (i == (num_retries - 1)) {
```

Comment:
maybe make it shorter, remove break and internal if block:
```
for (int i = 0; (i < num_retries) && (nullptr == recvbuf.get()); ++i) {
    try {
    } catch () {}
}

if (nullptr == recvbuf.get()) {
    FAIL() << "Failed to allocate buffer";
}

Reply:
done